src/dirtyfields/dirtyfields.py

```diff
@@ -129,7 +129,12 @@ def reset_state(sender, instance, **kwargs):
     if update_fields:
         for field_name in update_fields:
             field = sender._meta.get_field(field_name)
+
+            if field.get_attname() in instance.get_deferred_fields():
+                continue
```

Comment:
I'm not sure about this fix. We may just check that that the field name is in `new_state`?